server/src/modules/boards/boards.controller.ts

```diff
@@ -5,32 +5,46 @@ import {
   Get,
   Param,
   Post,
+  Query,
   UseGuards,
 } from '@nestjs/common';
 import { BoardsService } from './boards.service';
 import { CreateBoardDto } from './boards.dto';
 import { User } from '../auth/decorators/users.decorator';
 import { JwtAuthGuard } from '../auth/guards/jwt.auth.guard';
 import { BoardResponseObject } from '../../shared/interfaces/response-objects/BoardResponseObject';
+import { BoardsGetOption } from 'src/enums/boards.get.option';
+import { BoardsPaginatedResponseObject } from 'src/shared/interfaces/response-objects/BoardsPaginatedResponseObject';
 
 // TODO - verify permissions for endpointss
 
 @Controller('boards')
 export class BoardsController {
   constructor(private readonly boardsService: BoardsService) {}
 
-  // TODO - implement
   @Get()
   @UseGuards(JwtAuthGuard)
-  getUserBoards(@User('_id') userId: string): string {
-    return `User ${userId} boards IMPLEMENT ME`;
-    // return this.boardsService.getUserBoards(userId);
+  getUserRelatedBoards(
+    @User('_id') userId: string,
+    @Query('boardsGetOption') boardsGetOption: BoardsGetOption,
```

Comment:
Just minor thing... since this function is supposed to retrieve user-related boards by default (as its name suggests), perhaps we don't need the ALL_USER_RELATED option in the enum?

Reply:
Well, there are 3 options:

- owned by user
- shared for user
- all

I would stay with this I think

Reply:
For me, a query parameter acts as a filter; we provide it when we want specific information. The ALL option implies 'just retrieve everything,' so it's not really necessary in this case. If the function were more general and intended to retrieve all boards from our app, then this query parameter or a variation of it would be essential.

Reply:
Okay, I see now. I will correct this.

Reply:
Corrected